Kin-Rewards-Engine/Reputation-Model-Proposal.md

```diff
@@ -0,0 +1,44 @@
+
+
+## Proposal
+
+We believe a reputation system run by Kik is not only possible, but inevitable, at least so long as the system is in bootstrap mode and a more stable PoS mechanism is found. The reputation model is made possible due to one key observation, namely, that most users interacting with the Kin ecosystem will do it from the Kik messenger app (at least in the early days). Assuming there will be an option to get a Kin wallet within the app (or to incorporate an existing one with it), a link can be made between devices and Kin addresses. This mapping will allow a reputation model based on a one-device-one-vote basis. Here is a possible start of such a reputation system.
+
+It will apply to both users and developers (i.e., similarly to Uber's drivers' and passengers' reputation), which:
+
+1. Will be weighted during users' voting and calculation of developers' rewards.
+2. Will be used to incentivise behaviors we would like to promote, while punishing bad acting (reduce Sybil attacks, promote participation and spending, etc.).
+3. Will be individually visible to users (without the actual derivation, probably) in order to achieve the above.
+4. Everyone would be able to verify that the rewards are being distributed correctly and according to the pre-defined rules. We will periodically publish and explain what behavior increases/decreases the reputation, in order to incentivise good behavior, but won't necessarily share how is it calculated exactly.
+
+### Guidelines
+
+1. Developers will be identified entities. Their reputation will be calculated according to the quality of their KYC, the detail of their app plan, the marketing that they do, etc.
+2. Users are anonymous so this is more tricky. However, as explained earlier this can be mitigated as long as users link their wallets and participate in the ecosystem via the Kik messenger (and this is the key realization that makes reputation feasible).  Reputation will be given to each Kin address depending on things like:
+	* Whether the address is attached to the mobile app or not. In case it is, a lot of data is available - how long does the user have the app, how active is he, his age, occupation, etc. This can be calculated locally on the device and sent to a Kik server (to preserve privacy) or done on Kik's servers (if possible legally). E.g., the amount of addresses a certain device opened is super relevant to that user's reputation.
+ 	* Age of coins. The longer the coins sit in an address the more likely they are trustworthy.
+ 	* Age of address. E.g., a new address will have reputation 0 and will be completely ignored by the reward engine (unless it was opened in a wallet of a specific type of user). The age of the address will be calculated either using its creation time (when available; e.g., contracts) or when the first incoming transaction has happened.
+	* Interaction with many developers (an address interacting with plenty of developers is highly likely to be honest).
```

Comment:
Interactions can be faked using sybil attacks i.e. an attacker interacts with itself using 1000 fake accounts. We should take into account the reputation of the users this one is interacting with, and execute some weighted calculation to combat this attack vector.

Reply:
It's only an idea, but please note that it refers to interactions with developers and not regular users. In any case, interaction based reputation component will have to be Sybil resistant, otherwise it's missing the whole point.